src/appimagetool.c

```diff
@@ -457,6 +459,41 @@ bool readFile(char* filename, int* size, char** buffer) {
     return TRUE;
 }
 
+/* run a command outside the current appimage, block environs like LD_LIBRARY_PATH */
+int run_external(const char *filename, char *const argv []) {
+    int pid = fork();
+    if (pid == -1) {
+        g_print("run_external: fork failed");
+        // fork failed
+        return -1;
+    } else if (pid == 0) {
+        // blocks env defined in resources/AppRun
+        unsetenv("LD_LIBRARY_PATH");
+        unsetenv("PYTHONPATH");
+        unsetenv("XDG_DATA_DIRS");
+        unsetenv("PERLLIB");
+        unsetenv("GSETTINGS_SCHEMA_DIR");
+        unsetenv("QT_PLUGIN_PATH");
+        // runs command
+        execve(filename, argv, environ);
```

Comment:
You don't need to pass `environ` yourself, unless you want to specify your own, custom one. The parent environment will be used for child processes automatically.